tests/haskell-load-tests.el

```diff
@@ -83,3 +83,19 @@
    (search-forward "import Data.String")
    (haskell-goto-prev-error)
    (should (looking-at-p "Data.Mayb"))))
+
+(ert-deftest do-cabal-no-process ()
+  "Ensure that haskell-process-do-cabal will run cabal directly.
+
+In fact, we are really observing the message the function outputs
+immediately before invoking the command directly by looking for
+it in the *Messages* buffer---since the test environment doesn't
+have cabal installed, it's not really going to be able to run a
+cabal command.  We could look for the resulting error, but this
+should be both sufficient and resilient in case we ever do start
+testing with cabal laying around."
+  (haskell-process-do-cabal "help")
+  (with-current-buffer "*Messages*"
```

Comment:
Nice test, this is about what we want, to see that `haskell-process-do-cabal` does something.

There is a nice way to intercept deep calls with `el-mock`. We use it for example here: https://github.com/haskell/haskell-mode/blob/fe2cce799031abe76f7f6dc85d08ea2ba7b115e7/tests/haskell-process-tests.el#L31

Use `mocklet` to make sure execution path you want it really taken. Thanks!